pgbadger

```diff
@@ -1856,6 +1865,9 @@ Options:
     --exclude-time  regex  : any timestamp matching the given regex will be
 			     excluded from the report. Example: "2013-04-12 .*"
                              You can use this option multiple times.
+    --include-time  regex  : any timestamp not matching the given regex will be
```

Comment:
This should be included in README and doc/pgBadger.pod, right ? There might be a tool to update these copies.

Reply:
Ah yes, I had assumed the docs were gotten from here actually. Would you like me to add that?

Reply:
I'm not very confortable with the double negation in the sentence. But it's consistent with the one above.

Did you try to implement this feature with a negated regex ? Something like `--exclude-time '2013-(?!04-12 .*)'` ?

I'm wondering how the two options behave together. Do you have some hint on this ?

Reply:
@wesleybowman yes, in pgBadger, everything is commited *built-in*. So your PR should modify every files concerned. Whatever the way it is updated (manually or with a script like `tools/updt_embedded_rsc.pl`). Thanks !

Reply:
> Did you try to implement this feature with a negated regex ? Something like --exclude-time '2013-(?!04-12 .*)' ? 

I did not.

And I will add to the docs. I also am not a fan of the double negation, but I went for consistency. Maybe I can make that better as well

Reply:
I'm wondering if it's simpler to document the case with `--exclude-time '2017-(?!02.*)'` rather than adding a *sibling* option that might confuse UX.

Actually I would prefer a single `--time-regex` with behaviour of `--include-time`. So i prefer to add your option which i find more natural and maybe one time, we will deprecate the negation. Who knows ;-)

Reply:
I guess that a `=head1 USE CASE` would be nice in `docs/pgBadger.pod` for such documentation. What do you think of this ?

Reply:
> I guess that a =head1 USE CASE would be nice in docs/pgBadger.pod for such documentation. What do you think of this ?

I am really not a perl guy. I just dipped my toes in for this since I was using it a lot.

As for the negation, I think I found a better way to rephrase for the `--include-time`:
```
only timestamps matching the given regex will be included in the report. 
Example: "2013-04-12 .*"
```

But, if you are wanting the single command, then maybe this doesn't work

Reply:
I've tested the following:
```
pgbadger --exclude-time '2017-08-(?!31.*)'
```
and it indeed works.

If we do just want to go with this, then I would suggest adding an example to make it more immediately obvious that this is a possibility

Reply:
> I am really not a perl guy.

Me neither ! ^^

> I just dipped my toes in for this 

Kudo and thanks for the contribution !

> I think I found a better way to rephrase

Perfect.

> But, if you are wanting the single command, then maybe this doesn't work

Sorry if I looks to change my mind on every comment. I'm ok to merge this feature. I'm juste checking the completeness and consistency of this new option. I think it will be used for sure !

Reply:
> If we do just want to go with this, then I would suggest adding an example to make it more immediately obvious that this is a possibility

I had another experience with such behaviour. I'm really not a fan of double negation. So let's add `--include-time`, and see if one wins.

I prefer to document the simple negation `--include-time '2017-(?!02.*)'` rather than the double negation to whitelist date. This is why I even suggest to name it `--time-regex`.

Actually, there is already `--include-query` and `--exclude-query`. So go for `--include-time` too !

Reply:
:+1: I will push the documentation changes